src/ff7/field.cpp

```diff
@@ -586,48 +586,58 @@ void field_clip_with_camera_range_float(vector2<float>* point)
 {
 	field_trigger_header* field_triggers_header_ptr = *ff7_externals.field_triggers_header;
 	float half_width = 160;
-	if(is_fieldmap_wide())
+	auto camera_range = field_triggers_header_ptr->camera_range;
+
+	if(aspect_ratio == AR_WIDESCREEN && (is_fieldmap_wide() || widescreen.getMode() == WM_ZOOM))
 	{
-		int cameraRange = field_triggers_header_ptr->camera_range.right - field_triggers_header_ptr->camera_range.left;
-		half_width = ceil(wide_viewport_width / 4.f);
+		camera_range = widescreen.getCameraRange();
+
+		// This centers the background if necessary
+		int cameraRangeSize = camera_range.right - camera_range.left;
+		half_width = 160 + std::min(53, cameraRangeSize / 2 - 160);
```

Comment:
This doesn't seem to work well with fields that are already big enough. Half width shouldn't be depending on camera range. If camera range is 660, you are going to set `half_width` to 340, which is obviously going outside the background

Reply:
Oh my bad, it's the minimum of those two. But this can also be done at config.toml level by increasing the camera range right and left to 213 and -213

Reply:
Do we need some more work here then?

Reply:
It could be done in the config manually, but I think the centering it is better to do it through code because it would reduce the size of the config file and it would be easier to adapt it for other aspect ratios such as 16:10 in the future.

Reply:
For now, bigger aspect ratio such as 21:9 are not compatible in few spots. The hardest to fix is worldmap (i.e. clouds textures).

Reply:
Ok let's keep this as TODO for the future. But for now, we're good :)